app/src/main/java/com/github/lookupgroup27/lookup/model/map/Renderer.kt

```diff
@@ -13,45 +13,82 @@ import javax.microedition.khronos.opengles.GL10
  */
 class Renderer : GLSurfaceView.Renderer {
 
-  private lateinit var shapes: List<Object>
+  private lateinit var skyBox: SkyBox
 
   /** The camera used to draw the shapes on the screen. */
   val camera = Camera()
 
+  // Temporary storage for the MVP matrix
+  private val mvpMatrix = FloatArray(16)
+  private val viewMatrix = FloatArray(16)
+  private val projectionMatrix = FloatArray(16)
+
   override fun onSurfaceCreated(unused: GL10, config: EGLConfig) {
     // Set the background frame color
     GLES20.glClearColor(0.0f, 0.0f, 0.0f, 1.0f)
 
     GLES20.glEnable(GLES20.GL_DEPTH_TEST)
 
-    // Create the shapes (Make sure you always create the shapes after the OpenGL context is
-    // created)
-    shapes =
-        listOf(
-            Star(0.0f, 0f, -1f, floatArrayOf(1.0f, 1.0f, 1.0f)),
-            Star(0.24f, 0f, -0.97f, floatArrayOf(1.0f, 1.0f, 1.0f)),
-            Star(1f, 0f, 0f, color = floatArrayOf(1.0f, 0.0f, 0.0f)),
-            Star(0f, 1f, 0f, color = floatArrayOf(0.0f, 1.0f, 0.0f)),
-            Star(0f, 0f, 1f, color = floatArrayOf(0.0f, 0.0f, 1.0f)))
+    // Initialize the SkyBox
+    skyBox = SkyBox()
+    skyBox.initialize()
+
+    // Initialize the camera's matrices
+    Matrix.setIdentityM(camera.viewMatrix, 0)
+    Matrix.setIdentityM(projectionMatrix, 0)
   }
 
   override fun onDrawFrame(unused: GL10) {
-    // Redraw background color
-    GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT)
-    GLES20.glClear(GLES20.GL_DEPTH_BUFFER_BIT)
 
-    // Draw the shapes
-    for (shape in shapes) shape.draw(camera)
+    // Clear the screen
+    GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT or GLES20.GL_DEPTH_BUFFER_BIT)
+
+    // Calculate the MVP matrix (Model-View-Projection matrix) for the objects
+    Matrix.multiplyMM(mvpMatrix, 0, projectionMatrix, 0, camera.viewMatrix, 0)
+
+    /*// Render the SkyBox
+    skyBox.render(mvpMatrix)*/
+    // Create a modified view matrix for the skybox that ignores translations
+    val skyboxViewMatrix = FloatArray(16)
+    System.arraycopy(camera.viewMatrix, 0, skyboxViewMatrix, 0, 16)
+    // Set translation components to zero
+    skyboxViewMatrix[12] = 0f // X translation
+    skyboxViewMatrix[13] = 0f // Y translation
+    skyboxViewMatrix[14] = 0f // Z translation
```

Comment:
Actually, this code ensures that the skybox remains fixed relative to the camera's position, effectively creating the desired immersion. Without resetting the translation components, the skybox would move with the camera, which might result in the viewer perceiving the skybox as a static object instead of a surrounding environment.
The goal isn't just to account for camera movement but to avoid the illusion of the skybox being a static object in space. If we remove this code, the skybox will translate with the camera's position, breaking the immersive effect.